### ISSUE TITLE
'PRC' outputs 'Ethylene Gas' to input side for 'Hydrogen'

### ISSUE THREAD

**Issue by BaronNox** [aidancbrady/Mekanism#3190]
**Version: 1.7.10-8.1.9.265**
_YouTube link at bottom of post._

Imagine this setup:

BPT = Basic Pressurized Tube
PRC = Pressurized Reaction Chamber

'Electrolytic Seperator' splits H2O into 'Hydrogen' and 'Oxygen'.
'Oxygen' gets 'Excess Dumped' into a 'Gas Tank'.
'Hydrogen' goes into a 'Basic Pressurized Tube (=BPT)' which goes into a 'PRC' (pipeline length: 1).
'PRC' uses 'Biofuel' and 'Hydrogen' to create 'Ethylene'.
Sometimes Ethylene gets output to the same side the hydrogen comes in which results in a blocked 'BPT' which has to be manually replaced.
Side configs of 'PRC': Back is 'red' (input), right side is 'blue (gas output)'.
I tried everything plus google but there seems to be no way to filter or restrict a 'BPT' to a certain gas.
Imho this is a bug since the output config of the 'PRC' gets completly ignored.

P.s.: I'm relativly new to Mekanism. If I derped hard please go easy on me.

Naming: Mekanism items/blocks/options are surrounded by ' '.

As you can see in the following screenshots the 'BPT' contains 'Ethylene' not 'Hydrogen'.
'PRC's' front is pointing away from me in those screenshots.
Those 'Gas Tanks' on the left are empty.
![2016-04-27_22 22 21](https://cloud.githubusercontent.com/assets/15865490/14866975/8b53724e-0cc6-11e6-9639-173a1b1052f8.png)
![2016-04-27_22 23 11](https://cloud.githubusercontent.com/assets/15865490/14866996/a4e8171e-0cc6-11e6-8f1a-09749b90fe81.png)

Youtube link for those visual guys: https://www.youtube.com/watch?v=Gl9T3wpjFGo

Edit: Fixed typos, added screenshots, improved formatting.

**Comment by BaronNox** [comment]
Placing the 'Eloctrolyric Seperator' next to 'PRC' fixes blocking behaviour.

**Comment by aidancbrady** [comment]
I used the same setup you described and was unable to reproduce the issue.  Make sure that you didn't just accidentally connect a tube with the wrong gas.

I kept the setup running for about 5 minutes, the first time with a constant flow of oxygen from the Electrolytic Separator:

[](http://puu.sh/oyae8/9b7c05da71.png)

I then cut off the Electrolytic Separator and still no Ethylene was output to the wrong area:
[] (http://puu.sh/oyakL/02bd5bc173.png)

I had the exact same configuration as you, auto-ejecting to the gas tank, and wasn't experiencing any issues.  There's a good chance that I fixed this in v9.
